app/src/main/java/fr/free/nrw/commons/category/CategorizationFragment.java

```diff
@@ -320,8 +321,8 @@ protected void onPostExecute(ArrayList<String> result) {
                 latch.countDown();
             }
         };
-        prefixUpdaterSub.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
-        methodAUpdaterSub.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+        prefixUpdaterSub.executeOnExecutor(Executors.newSingleThreadExecutor());
+        methodAUpdaterSub.executeOnExecutor(Executors.newSingleThreadExecutor());
     }
```

Comment:
Yes you are right @whym , but if it solves the problem, we can say at least problem detection is right. Did you tried to reproduce the problem? 
For new solution approaches, we can already cancel tasks but when it is somewhere after onPeExecute, cancelling doesn't stop doItBackground and I am not sure if it is possible to do this somehow. My plan B is adding timeouts to doInBackgrounds to prevent hangs. But this can cause latency. Plus, I can try to using execute() method instead to make it serial. So controlling them would be a lot easier.